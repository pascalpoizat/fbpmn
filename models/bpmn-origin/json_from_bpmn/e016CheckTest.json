{
    "isInterrupt": {},
    "catN": {
        "Sender_": "Process",
        "Task_19wi6kk": "ReceiveTask",
        "StartEvent_1": "NoneStartEvent",
        "ExclusiveGateway_1gus05a": "XorGateway",
        "EndEvent_1d6yy2u": "NoneEndEvent",
        "EndEvent_19gcpb9": "NoneEndEvent",
        "Task_1cbrss1": "SendTask",
        "StartEvent_0x0o2ml": "NoneStartEvent",
        "ExclusiveGateway_1aidani": "EventBasedGateway",
        "EndEvent_1e5c16l": "NoneEndEvent",
        "Task_1c559zv": "SendTask",
        "EndEvent_198u4gq": "NoneEndEvent",
        "Receiver_": "Process",
        "Task_1d78ih0": "ReceiveTask"
    },
    "catE": {
        "SequenceFlow_1bot7ik": "NormalSequenceFlow",
        "MessageFlow_1n4m463": "MessageFlow",
        "SequenceFlow_0ob8kjq": "NormalSequenceFlow",
        "SequenceFlow_08o2r31": "NormalSequenceFlow",
        "SequenceFlow_01czfja": "NormalSequenceFlow",
        "SequenceFlow_0fxw5l8": "NormalSequenceFlow",
        "SequenceFlow_0cvzouu": "NormalSequenceFlow",
        "SequenceFlow_0tmjk0x": "NormalSequenceFlow",
        "SequenceFlow_1t3w3h5": "ConditionalSequenceFlow",
        "SequenceFlow_0qoid9s": "NormalSequenceFlow",
        "SequenceFlow_13vrdzd": "DefaultSequenceFlow",
        "MessageFlow_095vt4e": "MessageFlow"
    },
    "containN": {
        "Sender_": [
            "StartEvent_1",
            "ExclusiveGateway_1gus05a",
            "EndEvent_198u4gq",
            "EndEvent_1e5c16l",
            "Task_1c559zv",
            "Task_1cbrss1"
        ],
        "Receiver_": [
            "StartEvent_0x0o2ml",
            "EndEvent_1d6yy2u",
            "EndEvent_19gcpb9",
            "Task_1d78ih0",
            "Task_19wi6kk",
            "ExclusiveGateway_1aidani"
        ]
    },
    "messageE": {
        "MessageFlow_1n4m463": "m2",
        "MessageFlow_095vt4e": "m1"
    },
    "timeInformation": {},
    "targetE": {
        "SequenceFlow_1bot7ik": "ExclusiveGateway_1gus05a",
        "MessageFlow_1n4m463": "Task_19wi6kk",
        "SequenceFlow_0ob8kjq": "EndEvent_1d6yy2u",
        "SequenceFlow_08o2r31": "EndEvent_198u4gq",
        "SequenceFlow_01czfja": "Task_1d78ih0",
        "SequenceFlow_0fxw5l8": "ExclusiveGateway_1aidani",
        "SequenceFlow_0cvzouu": "EndEvent_1e5c16l",
        "SequenceFlow_0tmjk0x": "EndEvent_19gcpb9",
        "SequenceFlow_1t3w3h5": "Task_1c559zv",
        "SequenceFlow_0qoid9s": "Task_19wi6kk",
        "SequenceFlow_13vrdzd": "Task_1cbrss1",
        "MessageFlow_095vt4e": "Task_1d78ih0"
    },
    "name": "e016CheckTest",
    "attached": {},
    "containE": {
        "Sender_": [
            "SequenceFlow_1bot7ik",
            "SequenceFlow_1t3w3h5",
            "SequenceFlow_13vrdzd",
            "SequenceFlow_08o2r31",
            "SequenceFlow_0cvzouu"
        ],
        "Receiver_": [
            "SequenceFlow_0fxw5l8",
            "SequenceFlow_01czfja",
            "SequenceFlow_0qoid9s",
            "SequenceFlow_0ob8kjq",
            "SequenceFlow_0tmjk0x"
        ]
    },
    "edges": [
        "MessageFlow_095vt4e",
        "MessageFlow_1n4m463",
        "SequenceFlow_1bot7ik",
        "SequenceFlow_1t3w3h5",
        "SequenceFlow_13vrdzd",
        "SequenceFlow_08o2r31",
        "SequenceFlow_0cvzouu",
        "SequenceFlow_0fxw5l8",
        "SequenceFlow_01czfja",
        "SequenceFlow_0qoid9s",
        "SequenceFlow_0ob8kjq",
        "SequenceFlow_0tmjk0x"
    ],
    "messages": [
        "m1",
        "m2"
    ],
    "sourceE": {
        "SequenceFlow_1bot7ik": "StartEvent_1",
        "MessageFlow_1n4m463": "Task_1cbrss1",
        "SequenceFlow_0ob8kjq": "Task_1d78ih0",
        "SequenceFlow_08o2r31": "Task_1cbrss1",
        "SequenceFlow_01czfja": "ExclusiveGateway_1aidani",
        "SequenceFlow_0fxw5l8": "StartEvent_0x0o2ml",
        "SequenceFlow_0cvzouu": "Task_1c559zv",
        "SequenceFlow_0tmjk0x": "Task_19wi6kk",
        "SequenceFlow_1t3w3h5": "ExclusiveGateway_1gus05a",
        "SequenceFlow_0qoid9s": "ExclusiveGateway_1aidani",
        "SequenceFlow_13vrdzd": "ExclusiveGateway_1gus05a",
        "MessageFlow_095vt4e": "Task_1c559zv"
    },
    "nodes": [
        "Sender_",
        "Receiver_",
        "StartEvent_1",
        "ExclusiveGateway_1gus05a",
        "EndEvent_198u4gq",
        "EndEvent_1e5c16l",
        "Task_1c559zv",
        "Task_1cbrss1",
        "StartEvent_0x0o2ml",
        "EndEvent_1d6yy2u",
        "EndEvent_19gcpb9",
        "Task_1d78ih0",
        "Task_19wi6kk",
        "ExclusiveGateway_1aidani"
    ],
    "nameN": {
        "Task_19wi6kk": "B",
        "Task_1cbrss1": "B",
        "Task_1c559zv": "A",
        "Task_1d78ih0": "A"
    }
}