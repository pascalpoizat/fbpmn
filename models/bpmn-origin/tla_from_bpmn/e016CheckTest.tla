---------------- MODULE e016CheckTest ----------------

EXTENDS TLC, PWSTypes

VARIABLES nodemarks, edgemarks, net

Interest ==
  "Receiver_" :> { "m1", "m2" }
  @@ "Sender_" :> {  }

ContainRel ==
  "Receiver_" :> { "StartEvent_0x0o2ml", "EndEvent_1d6yy2u", "EndEvent_19gcpb9", "Task_1d78ih0", "Task_19wi6kk", "ExclusiveGateway_1aidani" }
  @@ "Sender_" :> { "StartEvent_1", "ExclusiveGateway_1gus05a", "EndEvent_198u4gq", "EndEvent_1e5c16l", "Task_1c559zv", "Task_1cbrss1" }

Node == { "Sender_", "Receiver_", "StartEvent_1", "ExclusiveGateway_1gus05a", "EndEvent_198u4gq", "EndEvent_1e5c16l", "Task_1c559zv", "Task_1cbrss1", "StartEvent_0x0o2ml", "EndEvent_1d6yy2u", "EndEvent_19gcpb9", "Task_1d78ih0", "Task_19wi6kk", "ExclusiveGateway_1aidani" }

Edge == { "MessageFlow_095vt4e", "MessageFlow_1n4m463", "SequenceFlow_1bot7ik", "SequenceFlow_1t3w3h5", "SequenceFlow_13vrdzd", "SequenceFlow_08o2r31", "SequenceFlow_0cvzouu", "SequenceFlow_0fxw5l8", "SequenceFlow_01czfja", "SequenceFlow_0qoid9s", "SequenceFlow_0ob8kjq", "SequenceFlow_0tmjk0x" }

Message == { "m1", "m2" }

msgtype ==
   "MessageFlow_095vt4e" :> "m1"
@@ "MessageFlow_1n4m463" :> "m2"


source ==
   "MessageFlow_095vt4e" :> "Task_1c559zv"
@@ "MessageFlow_1n4m463" :> "Task_1cbrss1"
@@ "SequenceFlow_1bot7ik" :> "StartEvent_1"
@@ "SequenceFlow_1t3w3h5" :> "ExclusiveGateway_1gus05a"
@@ "SequenceFlow_13vrdzd" :> "ExclusiveGateway_1gus05a"
@@ "SequenceFlow_08o2r31" :> "Task_1cbrss1"
@@ "SequenceFlow_0cvzouu" :> "Task_1c559zv"
@@ "SequenceFlow_0fxw5l8" :> "StartEvent_0x0o2ml"
@@ "SequenceFlow_01czfja" :> "ExclusiveGateway_1aidani"
@@ "SequenceFlow_0qoid9s" :> "ExclusiveGateway_1aidani"
@@ "SequenceFlow_0ob8kjq" :> "Task_1d78ih0"
@@ "SequenceFlow_0tmjk0x" :> "Task_19wi6kk"

target ==
   "MessageFlow_095vt4e" :> "Task_1d78ih0"
@@ "MessageFlow_1n4m463" :> "Task_19wi6kk"
@@ "SequenceFlow_1bot7ik" :> "ExclusiveGateway_1gus05a"
@@ "SequenceFlow_1t3w3h5" :> "Task_1c559zv"
@@ "SequenceFlow_13vrdzd" :> "Task_1cbrss1"
@@ "SequenceFlow_08o2r31" :> "EndEvent_198u4gq"
@@ "SequenceFlow_0cvzouu" :> "EndEvent_1e5c16l"
@@ "SequenceFlow_0fxw5l8" :> "ExclusiveGateway_1aidani"
@@ "SequenceFlow_01czfja" :> "Task_1d78ih0"
@@ "SequenceFlow_0qoid9s" :> "Task_19wi6kk"
@@ "SequenceFlow_0ob8kjq" :> "EndEvent_1d6yy2u"
@@ "SequenceFlow_0tmjk0x" :> "EndEvent_19gcpb9"

CatN ==
   "Sender_" :> Process
@@ "Receiver_" :> Process
@@ "StartEvent_1" :> NoneStartEvent
@@ "ExclusiveGateway_1gus05a" :> ExclusiveOr
@@ "EndEvent_198u4gq" :> NoneEndEvent
@@ "EndEvent_1e5c16l" :> NoneEndEvent
@@ "Task_1c559zv" :> SendTask
@@ "Task_1cbrss1" :> SendTask
@@ "StartEvent_0x0o2ml" :> NoneStartEvent
@@ "EndEvent_1d6yy2u" :> NoneEndEvent
@@ "EndEvent_19gcpb9" :> NoneEndEvent
@@ "Task_1d78ih0" :> ReceiveTask
@@ "Task_19wi6kk" :> ReceiveTask
@@ "ExclusiveGateway_1aidani" :> EventBased

CatE ==
   "MessageFlow_095vt4e" :> MessageFlow
@@ "MessageFlow_1n4m463" :> MessageFlow
@@ "SequenceFlow_1bot7ik" :> NormalSeqFlow
@@ "SequenceFlow_1t3w3h5" :> ConditionalSeqFlow
@@ "SequenceFlow_13vrdzd" :> DefaultSeqFlow
@@ "SequenceFlow_08o2r31" :> NormalSeqFlow
@@ "SequenceFlow_0cvzouu" :> NormalSeqFlow
@@ "SequenceFlow_0fxw5l8" :> NormalSeqFlow
@@ "SequenceFlow_01czfja" :> NormalSeqFlow
@@ "SequenceFlow_0qoid9s" :> NormalSeqFlow
@@ "SequenceFlow_0ob8kjq" :> NormalSeqFlow
@@ "SequenceFlow_0tmjk0x" :> NormalSeqFlow

LOCAL preEdges ==
  [ i \in {} |-> {}]
PreEdges(n,e) == preEdges[n,e]

PreNodes(n,e) == { target[ee] : ee \in preEdges[n,e] }
          \union { nn \in { source[ee] : ee \in preEdges[n,e] } : CatN[nn] \in { NoneStartEvent, MessageStartEvent } }

BoundaryEvent ==
  [ i \in {} |-> {}]

WF == INSTANCE PWSWellFormed
ASSUME WF!WellTyped
ASSUME WF!WellFormedness

ConstraintNode == TRUE \* none
ConstraintEdge == TRUE \* none
Constraint == TRUE     \* none
INSTANCE PWSConstraints
INSTANCE UserProperties
INSTANCE PWSSemantics

================================================================

