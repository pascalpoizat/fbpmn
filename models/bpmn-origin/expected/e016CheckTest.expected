Working in /tmp/e016CheckTest.yt0E6 with 1 worker(s)
[92mtransformation done[0m
<<"Processes=", 2, "Nodes=", 14, "Gateway=", 2, "SF=", 10, "MF=", 2>>
---------- Network01Bag ----------
[X] Prop01Type
     states=36 trans=53 depth=11
[X] Prop02Safe
     states=36 trans=53 depth=11
[X] Prop03Sound
     states=36 trans=53 depth=11
[X] Prop04MsgSound
     states=36 trans=53 depth=11
---------- Network02FifoPair ----------
[X] Prop01Type
     states=36 trans=53 depth=11
[X] Prop02Safe
     states=36 trans=53 depth=11
[X] Prop03Sound
     states=36 trans=53 depth=11
[X] Prop04MsgSound
     states=36 trans=53 depth=11
---------- Network03Causal ----------
[X] Prop01Type
     states=36 trans=53 depth=11
[X] Prop02Safe
     states=36 trans=53 depth=11
[X] Prop03Sound
     states=36 trans=53 depth=11
[X] Prop04MsgSound
     states=36 trans=53 depth=11
---------- Network04Inbox ----------
[X] Prop01Type
     states=36 trans=53 depth=11
[X] Prop02Safe
     states=36 trans=53 depth=11
[X] Prop03Sound
     states=36 trans=53 depth=11
[X] Prop04MsgSound
     states=36 trans=53 depth=11
---------- Network05Outbox ----------
[X] Prop01Type
     states=36 trans=53 depth=11
[X] Prop02Safe
     states=36 trans=53 depth=11
[X] Prop03Sound
     states=36 trans=53 depth=11
[X] Prop04MsgSound
     states=36 trans=53 depth=11
---------- Network06Fifo ----------
[X] Prop01Type
     states=36 trans=53 depth=11
[X] Prop02Safe
     states=36 trans=53 depth=11
[X] Prop03Sound
     states=36 trans=53 depth=11
[X] Prop04MsgSound
     states=36 trans=53 depth=11
---------- Network07RSC ----------
[X] Prop01Type
     states=36 trans=53 depth=11
[X] Prop02Safe
     states=36 trans=53 depth=11
[X] Prop03Sound
     states=36 trans=53 depth=11
[X] Prop04MsgSound
     states=36 trans=53 depth=11
done.
